SocialCareCaseViewerApi.Tests/V1/UseCase/Residents/GetAllUseCaseTests.cs

```diff
@@ -0,0 +1,185 @@
+        public void DoesNotCallGetResidentsBySearchCriteriaMethodWhenMosaicIdIsProvided()
+        {
+            var request = new ResidentQueryParam() { MosaicId = "43" };
+
+            _mockDatabaseGateway.Setup(x => x.GetResidentsBySearchCriteria(0, 10, null, null, null, null, null, null, null))
+              .Returns(new List<ResidentInformation>());
+
+            _mockDatabaseGateway.Setup(x => x.GetPersonIdsByEmergencyId(It.IsAny<long>())).Returns(new List<long>());
+
+            _getAllUseCase.Execute(request, cursor: 0, limit: 4);
+
+            _mockDatabaseGateway.Verify(x => x.GetResidentsBySearchCriteria(
+                It.IsAny<int>(),
+                It.IsAny<int>(),
+                It.IsAny<long>(),
+                It.IsAny<string>(),
+                It.IsAny<string>(),
+                It.IsAny<string>(),
+                It.IsAny<string>(),
+                It.IsAny<string>(),
+                It.IsAny<string>()),
+                Times.Never);
+        }
+
+        [Test]
+        public void CallsGetPersonByMosaicIdWhenMosaicIdIsProvided()
+        {
+            var request = new ResidentQueryParam() { MosaicId = "43" };
+
+            _mockDatabaseGateway.Setup(x => x.GetPersonIdsByEmergencyId(It.IsAny<long>())).Returns(new List<long> { 1, 2 });
+            _mockDatabaseGateway.Setup(x => x.GetPersonsByListOfIds(It.IsAny<List<long>>())).Returns(new List<Person>() { TestHelpers.CreatePerson() });
+
+            _getAllUseCase.Execute(request, cursor: 0, limit: 4);
+
+            _mockDatabaseGateway.Verify(x => x.GetPersonByMosaicId(Convert.ToInt64(request.MosaicId)));
+        }
+
+        [Test]
+        public void CallsGetPersonIdsByEmergencyIdWhenMosaicIdIsProvided()
+        {
+            var request = new ResidentQueryParam() { MosaicId = "43" };
+
+            _mockDatabaseGateway.Setup(x => x.GetPersonIdsByEmergencyId(It.IsAny<long>())).Returns(new List<long>() { 1, 2 });
+            _mockDatabaseGateway.Setup(x => x.GetPersonsByListOfIds(It.IsAny<List<long>>())).Returns(new List<Person>() { TestHelpers.CreatePerson() });
+
+            _getAllUseCase.Execute(request, cursor: 0, limit: 4);
+
+            _mockDatabaseGateway.Verify(x => x.GetPersonIdsByEmergencyId(Convert.ToInt64(request.MosaicId)));
+        }
+
+        [Test]
+        public void CallsGetPersonsByListOfIdsWhenMatchingEmergencyIdsFound()
+        {
+            var request = new ResidentQueryParam() { MosaicId = "43" };
+
+            var listOfMatchingIds = new List<long> { 1, 2 };
+
+            _mockDatabaseGateway.Setup(x => x.GetPersonIdsByEmergencyId(It.IsAny<long>())).Returns(listOfMatchingIds);
+            _mockDatabaseGateway.Setup(x => x.GetPersonsByListOfIds(It.IsAny<List<long>>())).Returns(new List<Person>() { TestHelpers.CreatePerson() });
+
+            _getAllUseCase.Execute(request, cursor: 0, limit: 4);
+
+            _mockDatabaseGateway.Verify(x => x.GetPersonsByListOfIds(listOfMatchingIds));
+        }
```

Comment:
What do you think about adding `TestCase`s for different `MosaicId`s? As the code mentions we want to be removing prefixes and other letters.

Reply:
I've added test cases for each